swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
 class Vendor:
-    pass
+
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory else []
+
+    def add(self, new_item):
+        """ Adds argument passed into new_item to Vendor's inventory list, and returns the new item. """
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item_to_remove):
+        """ Removes argument passed into item_to_remove from Vendor's inventory list, and returns the removed item. """
+        if item_to_remove not in self.inventory:
+            return False
+        self.inventory.remove(item_to_remove)
+        removed_item = item_to_remove
+        return removed_item
+
+    def get_by_category(self, category):
+        """ Returns a list of items that have the category attribute given argument for category """
+        inventory_items_in_category = [
+            item for item in self.inventory if item.category == category]
+        return inventory_items_in_category
+
+    def swap_items(self, trading_vendor, my_item, their_item):
+        """
+        Takes in 3 arguments: another vendor instance, an item in the calling vendor's inventory, and an item from the
+        the other vendor's inventory.
+        If the items are in their expected inventories, the two given items are traded, and we return True. 
+        Else if either of the items aren't in their expected inventories, we return False.
+        """
+        if my_item not in self.inventory or their_item not in trading_vendor.inventory:
+            return False
+
+        self.remove(my_item)
+        trading_vendor.add(my_item)
+        self.add(their_item)
+        trading_vendor.remove(their_item)
+        return True
+
+    def swap_first_item(self, trading_vendor):
+        """ 
+        Takes in 1 argument: another vendor instance. Trades the first item in the calling instance's inventory
+        with the first item in the other vendor's first item. Returns True.
+        """
+        if not self.inventory or not trading_vendor.inventory:
+            return False
+
+        my_first = self.inventory[0]
+        their_first = trading_vendor.inventory[0]
+        self.swap_items(trading_vendor, my_first, their_first)
+        return True
```

Comment:
This function is really clean! Additionally, we can use multiple assignment to swap the first items like so:
```suggestion
    def swap_first_item(self, trading_vendor):
        """ 
        Takes in 1 argument: another vendor instance. Trades the first item in the calling instance's inventory
        with the first item in the other vendor's first item. Returns True.
        """
        if not self.inventory or not trading_vendor.inventory:
            return False

        self.inventory[0], trading_vendor.inventory[0] = trading_vendor.inventory[0], self.inventory[0]
   
        return True
```

Reply:
The `swap_items` function contains a `remove` method, making the function's worst-case time complexity O(n). Multiple assignment instead will turn the time complexity of this entire function to O(1).